lib/datadog/appsec/component.rb

```diff
@@ -1,21 +1,31 @@
 # typed: ignore
 # frozen_string_literal: true
 
-require_relative 'assets'
 require_relative 'processor'
 
 module Datadog
   module AppSec
     # Core-pluggable component for AppSec
-    class Component
-      attr_reader :processor
 
-      def initialize(settings, agent_settings, tracer)
-        @processor = Processor.new
+    module Component
+      class Client
```

Comment:
Right, so the component is what gets added to the core components file, following the other libraries
https://github.com/DataDog/dd-trace-rb/blob/b7591eb03d626f5c85b5f5b9e74c530366b86242/lib/datadog/core/configuration/components.rb#L21-L23

Since the `Components` is a module, we can not instantiate it. 

That is why I created a separate class that holds the data for the other app-sec related things: processor, transport, etc... 

I couldn't figure out a better name, so I opted for having inside the component module for know. Happy to find a better name for it. 

Any suggestions? @lloeki

Reply:
Ah, I missed that you changed `Component` to be a module.

Actually the others should become classes, but that's part of a larger scoped refactoring that we're delaying for now, with AppSec leading the design, hence why `AppSec::Component` should be a class here.

Reply:
Here's a ref: https://github.com/DataDog/dd-trace-rb/pull/2561#discussion_r1084216245

The idea being that component-specific instances should be owned by each namespace and live inside each component instead of inside `Core::Configuration::Components`.